benchexec/tools/jayhorn.py

```diff
@@ -0,0 +1,76 @@
+"""
+BenchExec is a framework for reliable benchmarking.
+This file is part of BenchExec.
+Copyright (C) 2007-2015  Dirk Beyer
+All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import logging
+import xml.etree.ElementTree as ET
+import os
+import shutil
+import zipfile
+
+import benchexec.util as util
+import benchexec.tools.template
+import benchexec.result as result
+
+class Tool(benchexec.tools.template.BaseTool):
+    """
+    Tool info for JayHorn (https://github.com/jayhorn/jayhorn).
+    JayHorn does not support jar files yet;
+    we therefore unpack the benchmark jar file.
+    """
+
+    REQUIRED_PATHS = [
+                  "jayhorn.jar"
+                  "jayhorn" # contains: java -jar jayhorn.jar
+                  ]
+    def executable(self):
+        return util.find_executable('jayhorn')
+
+
+    def version(self, executable):
+        return "5.1" # --version not supported by JayHorn
```

Comment:
Tool-info modules should never hard-code version numbers. Otherwise users who benchmark a newer version of your tool with BenchExec will get wrong version information, which can lead to wrong conclusions.

If your tool does not output a version number, please do not add the `version` method at all.